Automattic/jetpack
# SAL: Syncing class.json-api-site-base.php with WordPress.com.

#### Changes proposed in this Pull Request:

* As part of the de-fusioning efforts, this PR sync changes made in the WordPress.com `class.json-api-site-base.php` file. This includes adding several functions or function declarations. It also includes the addition of an abstract function, `get_videopress_storage_used`, which required the function to be added to `class.json-api-site-jetpack.php` (added with a default return).
* The WordPress.com file is already 'in parity' except for one function - `get_site_goals` - which exists in the Jetpack file only. I haven't added that function to WordPress.com as it will exist in the new  `jetpack-plugin` production file once this is merged and packages updated, anyway.

#### Other information:


#### Jetpack product discussion

Linked issue: #25202


#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* You can test that the changes don't cause any endpoint breakage by using the WordPress.com developer console to test the endpoint , using the WP.COM API GET request for /sites/{blog_id}/ for example. A breaking change will affect the output, whereas it should work fine otherwise.
* You can also test features that may be impacted by this file, or adding an error log in a function within that main `class.json-api-site-base.php` file (such as `user_can_view_post` - and in that case then visiting WordPress.com and visiting the WordPress.com post editor - you should see your error log output then).
